src/Adapter/ContextLoadHelper.php

```diff
@@ -0,0 +1,140 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter;
+
+use Context;
+use Currency;
+use DummyController;
+use Employee;
+use Shop;
+
+/**
+ * Helps loading specific context
+ */
+class ContextLoadHelper
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @param LegacyContext $context
+     */
+    public function __construct(LegacyContext $context)
+    {
+        $this->context = $context->getContext();
+    }
+
+    public function loadGenericContext(
+        ?string $controllerClassName = null,
+        ?int $currencyId = null,
+        ?int $employeeId = null,
+        ?int $shopId = null,
+        ?int $shopGroupId = null)
+    {
+        $this->loadCurrencyContext($currencyId);
+        $this->loadControllerContext($controllerClassName);
+        $this->loadEmployeeContext($employeeId);
+
+        if (null !== $shopId) {
+            $this->loadShopContext($shopId);
+        }
+        if (null !== $shopGroupId) {
+            $this->loadShopGroupId($shopGroupId);
+        }
+    }
+
+    public function loadControllerContext(?string $controllerClassName = null)
+    {
+        if (null === $controllerClassName) {
+            $this->context->controller = new DummyController();
+
+            return;
+        }
+
+        if (!class_exists($controllerClassName)) {
+            throw new \RuntimeException(
+                sprintf(
+                    'Cannot load controller context for classname %s',
+                    $controllerClassName
+                )
+            );
+        }
+
+        $this->context->controller = new $controllerClassName();
+    }
+
+    public function loadCurrencyContext(?int $currencyId = null)
+    {
+        if (null === $currencyId) {
+            $currency = new Currency();
+            $currency->precision = 2;
+            $this->context->currency = $currency;
+
+            return;
+        }
+
+        $this->context->currency = new Currency($currencyId);
+    }
+
+    public function loadEmployeeContext(?int $employeeId = null)
+    {
+        if (null === $employeeId) {
+            $this->context->employee = new Employee();
+
+            return;
+        }
+
+        $this->context->employee = new Employee($employeeId);
+    }
+
+    public function loadShopContext(?int $shopId = null)
+    {
+        if (null === $shopId) {
+            $this->context->shop = new Shop();
+            Shop::setContext(Shop::CONTEXT_SHOP, 1);
+
+            return;
+        }
+
+        $this->context->shop = new Shop($shopId);
+        Shop::setContext(Shop::CONTEXT_SHOP, $shopId);
+    }
```

Comment:
```suggestion
    public function loadShopContext(int $shopId = 1)
    {
        $this->context->shop = new Shop($shopId);
        Shop::setContext(Shop::CONTEXT_SHOP, $shopId);
    }
```

Regarding install.php, and other things like that, we never use empty shop :thinking: